api/provider/aws/job_runner.go

```diff
@@ -68,189 +69,196 @@ func (r *JobRunner) Run(j models.Job) error {
+func (r *JobRunner) createEnvironment(jobID, request string) (string, error) {
 	var req models.CreateEnvironmentRequest
 	if err := json.Unmarshal([]byte(request), &req); err != nil {
-		return errors.New(errors.InvalidRequest, err)
+		return "", errors.New(errors.InvalidRequest, err)
 	}
 
 	environment, err := r.environmentProvider.Create(req)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	return r.jobStore.SetJobResult(jobID, environment.EnvironmentID)
+	return environment.EnvironmentID, nil
 }
 
-func (r *JobRunner) createLoadBalancer(jobID, request string) error {
+func (r *JobRunner) createLoadBalancer(jobID, request string) (string, error) {
 	var req models.CreateLoadBalancerRequest
 	if err := json.Unmarshal([]byte(request), &req); err != nil {
-		return errors.New(errors.InvalidRequest, err)
+		return "", errors.New(errors.InvalidRequest, err)
 	}
 
 	loadBalancer, err := r.loadBalancerProvider.Create(req)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	return r.jobStore.SetJobResult(jobID, loadBalancer.LoadBalancerID)
+	return loadBalancer.LoadBalancerID, nil
 }
 
-func (r *JobRunner) createService(jobID, request string) error {
+func (r *JobRunner) createService(jobID, request string) (string, error) {
 	var req models.CreateServiceRequest
 	if err := json.Unmarshal([]byte(request), &req); err != nil {
-		return errors.New(errors.InvalidRequest, err)
+		return "", errors.New(errors.InvalidRequest, err)
 	}
 
 	service, err := r.serviceProvider.Create(req)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	return r.jobStore.SetJobResult(jobID, service.ServiceID)
+	// scale up after the service has been added into the ecs service scheduler
+	r.scaler.ScheduleRun(req.EnvironmentID)
+	return service.ServiceID, nil
 }
 
-func (r *JobRunner) createTask(jobID, request string) error {
+func (r *JobRunner) createTask(jobID, request string) (string, error) {
 	var req models.CreateTaskRequest
 	if err := json.Unmarshal([]byte(request), &req); err != nil {
-		return errors.New(errors.InvalidRequest, err)
+		return "", errors.New(errors.InvalidRequest, err)
 	}
 
-	task, err := r.taskProvider.Create(req)
-	if err != nil {
-		return err
-	}
+	// scale up prior to creating the task so we have room in the cluster
+	r.scaler.ScheduleRun(req.EnvironmentID)
+
+	return catchAndRetry(time.Hour*24, func() (result string, err error, shouldRetry bool) {
+		log.Printf("[DEBUG] [JobRunner] Creating task %s", req.TaskName)
+		task, err := r.taskProvider.Create(req)
+		if err != nil {
+			log.Printf("[DEBUG] [JobRunner] Failed to create task %s: %v", req.TaskName, err)
+			return "", err, true
+		}
 
-	return r.jobStore.SetJobResult(jobID, task.TaskID)
+		return task.TaskID, nil, false
+	})
 }
 
-func (r *JobRunner) deleteDeploy(jobID, deployID string) error {
-	return r.deployProvider.Delete(deployID)
+func (r *JobRunner) deleteDeploy(jobID, deployID string) (string, error) {
```

Comment:
@sesh-kebab 
> Why don't deletion jobs update the Job Store?

We only populate the `result string` field when we create things -> it's the entity id of the new object.